File: 709. To Lower Case.py
# Implement function ToLowerCase() that has a string parameter str, and returns the same string in lowercase.

class Solution:
    def toLowerCase(self, str: str) -> str:
        s = [ord(c) for c in str]
        
        for i, c in enumerate(s):
            print(c)
            if 65 <= c <= 90:  # if ASCII uppercase
                s[i] = chr(s[i] + 32)
            else:
                s[i] = chr(c)
                
        return ''.join(s)

File: 709. To Lower Case.rs
// Implement function ToLowerCase() that has a string parameter str, and returns the same string in lowercase.

impl Solution {
    pub fn to_lower_case(str: String) -> String {  
        let mut buffer = vec![];
        
        for c in str.as_bytes().into_iter() {
            if (65 <= *c) && (*c <= 90) {
                buffer.push(c + 32);
            } else {
                buffer.push(*c);
            }
        }

    String::from_utf8(buffer).expect("Found invalid UTF-8")
    }
}
